File: LV1/소수 찾기.js
function solution(n) {
  let sieve = new Array(n + 1).fill(true);
  let m = Math.floor(Math.sqrt(n));
  for (let i = 2; i <= m; i++) {
    if (sieve[i]) {
      for (let j = i * i; j <= n; j += i) {
        sieve[j] = false;
      }
    }
  }
  let count = 0;
  for (let i = 2; i <= n; i++) {
    if (sieve[i]) count++;
  }
  return count;
}

File: LV1/소수 찾기.java
class Solution {

  public int solution(int n) {
    int answer = 0;
    boolean[] prime = new boolean[n + 1];
    for (int i = 2; i <= n; i++) {
      prime[i] = true;
    }
    for (int i = 2; i <= n; i++) {
      if (prime[i]) {
        for (int j = i * 2; j <= n; j += i) {
          prime[j] = false;
        }
      }
    }
    for (int i = 2; i <= n; i++) {
      if (prime[i]) {
        answer++;
      }
    }
    return answer;
  }
}
